src/builder.rs

```diff
@@ -0,0 +1,55 @@
+use crate::{errors::Error, Layer, Packet};
+
+#[derive(Debug, Default)]
+pub struct PacketBuilder {
+    inner: Packet,
+}
+
+impl PacketBuilder {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn stack(mut self, layer: Box<dyn Layer + Send>) -> Result<Self, Error> {
+        if self.inner.unprocessed.is_empty() {
+            self.inner.layers.push(layer);
+            Ok(self)
+        } else {
+            Err(Error::SculptingError(
+                "Cannot push layer on top of raw byte layer".to_string(),
+            ))
+        }
+    }
+
+    pub fn stack_bytes(mut self, bytes: Vec<u8>) -> Self {
+        self.inner.unprocessed = bytes;
```

Comment:
Should do a `mem::replace` here to make sure the current unprocessed if any are dropped.

Reply:
The current unprocessed is first dropped, and bytes is moved to unprocessed.